cmd/thanos/config.go

```diff
@@ -0,0 +1,60 @@
+package main
+
+import (
+	"net/url"
+	"time"
+
+	"github.com/prometheus/common/model"
+	"github.com/thanos-io/thanos/pkg/extflag"
+	thanosmodel "github.com/thanos-io/thanos/pkg/model"
+)
+
+type grpcConfig struct {
+	bindAddress    *string
```

Comment:
I think when we should have the values in the config rather than pointers. It'd be cleaner in that way, otherwise, we'd end up lots of value extactions.
We can leave pointers if they're optional.